### DISCUSSION TITLE
Overuse of Content keyword is often confusing.

### DISCUSSION THREAD

**Discussion by brunoAltinet** [OrchardCMS/OrchardCore#14240] [Ideas]
One thing i find a bit disorienting in orchard is how often the word Content is used for all sorts of things, and often for stuff that really is unrelated.

For example, in a view i would often use:
Model.Content  - this Content here is not actually the content, it's a zone (like Header and Footer). This should have been Model.Main

Then on some other part of the view i would see ContentItem.Content - this content is a dynamic which returns json values. This should have been named ContentItem.Value or ContetItem.Record of  ContetItem.JObject. You can easily see both of these contents in the same file and they would present completely different things.

There's a couple of other things which give me headaches, like ContentManager.Update(contentItem) not actually updating the database but the underlying json, so that name is  also misleading, but i've learned to live with those (eg. wrap them)

I think these naming changes would be a great help for eliminating confusion, especially when you're working inside liquid.
And they could also be done with fall back to previous naming.

**Comment by ns8482e** [comment]
IMO,

- `Model.Content` for zone shape is appropriate as normally in CMS zones are defined as Header,Footer,Content, Aside etc

- `ContentItem.Content` is also appropriate as it defines the Json content of the content item

- Content and ContentItem also appropriate as they are logical entity of Content management System.

Thing to remember ContentItem.Content is json

**Reply by brunoAltinet** [reply]
- You could argue that Main is used as often as Content for zones. Considering how overloaded and overused Content keyword is used in Orchard, Main would be easier to discern at a glance.
- Then why not call it just "Json" then? 
- I'm ok with that

**Comment by Piedone** [comment]
I see your points, and some of these I personally got confused about before too... Could you open a PR with some renames you find the most important? I can work with you to introduce them in a non-breaking way. Then, you can better present these to the community for feedback.

**Comment by ns8482e** [comment]
- Changing ContentItem.Content to ContentItem.Json is a breaking change 

- So does the changing Model.Content to Model.Main will introduce breaking changes

**Comment by brunoAltinet** [comment]
We can keep old props around, just mark the obsolete. I'll make a PR when i get some time.

**Reply by Piedone** [reply]
Yeah, that would be the basic idea. Thank you!
